controllers/subscriptions.go

```diff
@@ -21,6 +21,7 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
```

Comment:
This is an "experimental" package in go1.19, but it made it into the core go packages in go1.21. It contains the `slices.Contains()` method I use on line 91 below. So its not realllly experimental since it does make it into the core packages in a later go version, but normally I would not wanna use something like this in production code. Here I did since we will presumably upgrade to go1.21 at some point and this method will be available in the standard lib. However if anyone thinks we should refrain from using anything from the experimental package, I am ok with that and can remove it

go docs on the experimental subrepo: https://pkg.go.dev/golang.org/x/exp#section-readme

Reply:
gonna leave this as is - we talked about it in slack and decided its ok to use for the following reasons:
1) this package is coming from the go devs, not other parties
2) its likely "experimental" for a bit so they can adjust the interface without a breaking change
3) it is a very simple method
4) it was added to the go standard lib in go1.21 as is, so we assume its safe to use
5) dont reinvent the wheel :)